test/Sema/access-level-import-inconsistencies.swift

```diff
@@ -0,0 +1,66 @@
+/// Import inconsistencies.
+///
+/// Swift 5 case, with slow adoption of access level on imports.
+/// Report default imports if any other import of the same module
+/// has an access level below public (the default).
+///
+/// Swift 6 case, with default imports as internal.
+/// Don't report any mismatch.
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// REQUIRES: asserts
+
+/// Build the library.
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -o %t
+
+/// Check diagnostics.
+
+//--- Lib.swift
+public struct LibType {}
+
+// RUN: %target-swift-frontend -typecheck %t/OneFile_AllExplicit.swift -I %t \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify
+//--- OneFile_AllExplicit.swift
+public import Lib
```

Comment:
I don't think you need to do anything differently in this PR, but I do find it odd that we would accept conflicting access levels for imports of a single module in a single file and think we should reconsider this in the future. To me conflicting explicit imports in the same file seem likely to be indicative of a mistake that we should at least warn about.

Reply:
Good point, I would see the use of a per-file check for inconsistencies.